File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int letters = 0;
    int word = 0;
    int sentences = 0;

    string text = get_string("Text: ");

    // count letters
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalnum(text[i]))
        {
            letters++;
        }
    }
    // printf("%i letter(s)\n", letters);

    //count words
    for (int j = 0; j < strlen(text); j++)
    {
        if (isspace(text[j]))
        {
            word++;
        }
    }
    int words = word + 1;
    // printf("%i word(s)\n", words);

    //count sentences
    for (int k = 0; k < strlen(text); k++)
    {
        if (text[k] == '.' || text[k] == '!' || text[k] == '?')
        {
            sentences++;
        }
    }
    // printf("%i sentence(s)\n", sentences);

    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: pset6/readability/readability.py
letters = 0
words = 1
sentences = 0

text = input("Text: ")

for i in range(0, len(text), 1):
    if text[i].isalpha():
        letters += 1
    if text[i].isspace():
        words += 1
    if text[i] == "." or text[i] == "!" or text[i] == "?":
        sentences += 1

l = letters / words * 100
s = sentences / words * 100
index = round(0.0588 * l - 0.296 * s - 15.8)

if (index < 1):
    print("Before Grade 1")
elif (1 <= index <= 16):
    print("Grade %d" % index)
else:
    print("Grade 16+")